serial_communication/web/routes/serial_handler.py

```diff
@@ -70,62 +71,62 @@ def send_ngrok_link(target_port=None):
         if public_url:
             print(f"Ngrok URL is available: {public_url}")
             send_to_serial_port("sms " + public_url)
-            ngrok_link = public_url
-            global ngrok_link_sent
-            ngrok_link_sent = True
+            CURRENT_NGROK_LINK = public_url
+            NGROK_LINK_SENT = True
             # Perform other tasks with ngrok_url
         else:
             print("Failed to obtain Ngrok URL.")
             send_to_serial_port("sms Failed to obtain Ngrok URL.")
-    except Exception as e:
+    except Exception as e: # pylint: disable=broad-except
         print("NGROK not initialized")
         write_in_file("ngrok_logger", "\nERROR:\n" + str(e))
 
 
 def update_namaz_time():
+    """Will update update namaz time on TTGO-TCall"""
     try:
-        global current_time
+        current_time = ""
         # Replace this URL with the actual URL of the prayer times for Lahore
         url = os.getenv("LAHORE_NAMAZ_TIME")
 
         # Fetch the HTML content of the webpage
-        response = requests.get(url)
+        response = requests.get(url, timeout=10)
         soup = BeautifulSoup(response.content, 'html.parser')
 
         # Fetch the current time online
         try:
             response = requests.get(
-                'http://worldtimeapi.org/api/timezone/Asia/Karachi')
+                'http://worldtimeapi.org/api/timezone/Asia/Karachi', timeout=10)
             data = response.json()
             current_time = datetime.datetime.fromisoformat(data['datetime'])
             current_time = current_time.strftime("%I:%M %p")
             # return "10:00 AM"
         except requests.exceptions.RequestException as e:
             exception_logger("part_of_update_namaz_time", e)
-            return
+            return None
 
         if not current_time:
             print("Failed to fetch current time online.")
-            return
-        # else:
-            # print(f"Current time: {current_time}")
+            return None
 
         # Find the prayer time row that corresponds to the next prayer after the current time
         prayer_times = soup.find_all('td', {'data-label': True})
-        next_prayer = None
+        next_prayer_name = None #refer to prayer name
+        next_prayer_time = None
+
+        for prayer_time in prayer_times:
+            prayer_time = prayer_time.text.strip()
+            if (datetime.datetime.strptime(prayer_time, '%I:%M %p') >
+                    datetime.datetime.strptime(current_time, '%I:%M %p')):
 
-        for time in prayer_times:
-            prayer_time = time.text.strip()
-            if datetime.datetime.strptime(prayer_time, '%I:%M %p') > datetime.datetime.strptime(current_time, '%I:%M %p'):
-                next_prayer = time['data-label']
+                next_prayer_name = prayer_time['data-label']
+                next_prayer_time = prayer_time
                 break
```

Comment:
This part don't seems fine, please verify at you end.

Reply:
Yes it leads to exception i have solved this issue you can merge PR now.